pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.pinot.controller.helix.core.minion.generator.PinotTaskGenerator;
+import org.apache.pinot.controller.helix.core.minion.generator.TaskGeneratorUtils;
+import org.apache.pinot.core.common.MinionConstants;
+import org.apache.pinot.core.minion.PinotTaskConfig;
+import org.apache.pinot.spi.annotations.minion.TaskGenerator;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableTaskConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@TaskGenerator
+public class PurgeTaskGenerator implements PinotTaskGenerator {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(PurgeTaskGenerator.class);
+
+    private static final PurgeTaskGenerator INSTANCE = new PurgeTaskGenerator();
+
+    protected ClusterInfoAccessor _clusterInfoAccessor;
+
+    public static Object newInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public void init(ClusterInfoAccessor clusterInfoAccessor) {
+        LOGGER.info("I'm in init for the purge");
+        _clusterInfoAccessor = clusterInfoAccessor;
+    }
+
+
+    @Override
+    public String getTaskType() {
+        return MinionConstants.PurgeTask.TASK_TYPE;
+    }
+
+
+
+    @Override
+    public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
+        String taskType = MinionConstants.PurgeTask.TASK_TYPE;
+        List<PinotTaskConfig> pinotTaskConfigs = new ArrayList<>();
+        Set<Segment> runningSegments =
+                TaskGeneratorUtils.getRunningSegments(MinionConstants.PurgeTask.TASK_TYPE, _clusterInfoAccessor);
+        for (TableConfig tableConfig : tableConfigs) {
+            String tableName = tableConfig.getTableName();
+            TableTaskConfig tableTaskConfig = tableConfig.getTaskConfig();
+            Preconditions.checkNotNull(tableTaskConfig);
+            Map<String, String> taskConfigs =
+                    tableTaskConfig.getConfigsForTaskType(MinionConstants.PurgeTask.TASK_TYPE);
+            Preconditions.checkNotNull(taskConfigs, "Task config shouldn't be null for Table: {}", tableName);
+
+            LOGGER.info("Start generating task configs for table: {} for task: {}", tableName, taskType);
+
+            if (tableConfig.getTableType() == TableType.REALTIME) {
+                LOGGER.info("Task type : {}, cannot be run on table of type  {}",
+                        taskType, TableType.REALTIME);
+                continue;
+            }
+                // Get max number of tasks for this table
+            int tableMaxNumTasks;
+            String tableMaxNumTasksConfig = taskConfigs.get(MinionConstants.TABLE_MAX_NUM_TASKS_KEY);
+            if (tableMaxNumTasksConfig != null) {
+                try {
+                    tableMaxNumTasks = Integer.parseInt(tableMaxNumTasksConfig);
+                } catch (Exception e) {
+                    tableMaxNumTasks = Integer.MAX_VALUE;
```

Comment:
Print a log here to warn that the value is incorrectly set?

Reply:
Done in the comming commit